o3de/o3de
# Intermediate assets that have had their source removed are no longer processed.

Signed-off-by: AMZN-stankowi <4838196+AMZN-stankowi@users.noreply.github.com>

## What does this PR do?

Before this change:
If the source file for an intermediate asset was removed, and changes were made outside that intermediate asset in a way that would cause it to fail to process, the intermediate asset would fail to process instead of just being removed/ignored like it should be.

To test this bug, follow these steps:
1) Sync to 955f2b2028914fa38ded66e7dcf11cf81a76fd48
2) Build & process assets
3) Sync to 4f917ff7561db3f454be1141185c11f6d493a59a (the material update)
4) Build & process assets
5) Wait for all assets to finish processing - some assets, like FBX files, will fail at first, but may succeed later.
6) Notice that some assets are still marked as failed to process, even when the Asset Processor is idle.
7) Look at those specific assets: Notice they are intermediate assets.
8) Look for the source for those intermediate assets: Notice they are missing.
9) Close Asset Processor
10) Open it again
11) Notice all assets process correctly now

This change adds an extra step in job evaluation: If the file is an intermediate asset, the file operation is not a delete operation, it's not a folder, then this looks up the source for the intermediate asset, and verifies the source still exists on disk. If not, it skips processing this intermediate asset. Other cleanup happens later, this removes that initial failure that causes Jenkins to get stuck and would confuse people on first run of Asset Processor.

## How was this PR tested?

Manual testing with the above steps in several variations.

Ran existing and new automated tests.
